tun/checksum.go

```diff
@@ -0,0 +1,42 @@
+package tun
+
+import "encoding/binary"
+
+// TODO: Explore SIMD and/or other assembly optimizations.
```

Comment:
Wondering - do you ever see this function coming up when profiling? What priority should that kind of optimization have if we're going to do it sometime down the road?

Reply:
I've observed this func accounting for ~6% of samples via perf. UDP i/o accounted for >20% in the same capture. Worth optimizing eventually, but not the highest priority IMO.

Reply:
Huh, okay. Noted.